python-stdlib/time/time.py

```diff
@@ -0,0 +1,54 @@
+import re
+
+_TS_YEAR = const(0)
+_TS_MON = const(1)
+_TS_MDAY = const(2)
+_TS_HOUR = const(3)
+_TS_MIN = const(4)
+_TS_SEC = const(5)
+_TS_WDAY = const(6)
+_TS_YDAY = const(7)
+_TS_ISDST = const(8)
+
+_WDAY = const(("Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"))
+_MDAY = const(
+    (
+        "January",
+        "February",
+        "March",
+        "April",
+        "May",
+        "June",
+        "July",
+        "August",
+        "September",
+        "October",
+        "November",
+        "December",
+    )
+)
+
+_time_regex = None
+
+
+def strftime(datefmt, ts):
+    global _time_regex
+    if _time_regex is None:
+        _time_regex = re.compile(r"%(-?[a-zA-Z])")
+    return _time_regex.sub(r"%(\1)s", datefmt) % {
+        "a": _WDAY[ts[_TS_WDAY] - 1][0:3],
+        "A": _WDAY[ts[_TS_WDAY] - 1],
+        "b": _MDAY[ts[_TS_MON] - 1][0:3],
+        "B": _MDAY[ts[_TS_MON] - 1],
+        "d": f"{ts[_TS_MDAY]:02d}",
```

Comment:
I think it's better to stick with `%` formatting here, ie `"%02d" % ts[_TS_MDAY]`.  Because this function already uses `%` formatting to do the final formatting step, and it'll mean it can work on ports that don't have f-strings enabled.